stubs/networkx/networkx/classes/digraph.pyi

```diff
@@ -0,0 +1,29 @@
+from _typeshed import Incomplete
+from collections.abc import Iterator
+from typing import Generic
+
+from networkx.classes.coreviews import AdjacencyView
+from networkx.classes.graph import Graph, _Node
+from networkx.classes.reportviews import (
+    InDegreeView,
+    InEdgeView,
+    InMultiDegreeView,
+    OutDegreeView,
+    OutEdgeView,
+    OutMultiDegreeView,
+)
+
+class DiGraph(Graph[_Node], Generic[_Node]):
+    succ: AdjacencyView[_Node, _Node, dict[str, Incomplete]]
+    pred: AdjacencyView[_Node, _Node, dict[str, Incomplete]]
+    def has_successor(self, u: _Node, v: _Node) -> bool: ...
+    def has_predecessor(self, u: _Node, v: _Node) -> bool: ...
+    def successors(self, n: _Node) -> Iterator[_Node]: ...
+    def predecessors(self, n: _Node) -> Iterator[_Node]: ...
+    in_edges: InEdgeView[_Node]
+    in_degree: InDegreeView[_Node] | InMultiDegreeView[_Node]  # ugly hack to make MultiDiGraph work
```

Comment:
Just tried and it would lead to `error: The erased type of self "networkx.classes.multigraph.MultiDiGraph[Any]" is not a supertype of its class "networkx.classes.graph.DiGraph[_Node`1]"  [misc]` anyway. Same with `def is_directed` and `def is_multigraph` which could've been dealt with this way.

I'll leave this as-is for now.